tests/suites/test_suite_psa_crypto.function

```diff
@@ -6951,19 +6956,107 @@ void derive_output( int alg_arg,
+                                        PSA_TLS12_PSK_TO_MS_PSK_MAX_SIZE );
+                        }
+
+                        PSA_ASSERT( psa_key_derivation_input_key(
+                                        &operation, steps[i], keys[i] ) );
+                        break;
+                    default:
+                        TEST_ASSERT( ! "default case not supported" );
+                        break;
+                }
+                break;
+            case PSA_KEY_DERIVATION_INPUT_OTHER_SECRET:
+                switch ( other_key_input_type )
                 {
-                    PSA_ASSERT( psa_get_key_attributes( keys[i], &attributes ) );
-                    TEST_ASSERT( PSA_BITS_TO_BYTES( psa_get_key_bits( &attributes ) ) <=
-                                 PSA_TLS12_PSK_TO_MS_PSK_MAX_SIZE );
+                    case 0: // input bytes
+                        TEST_EQUAL( psa_key_derivation_input_bytes(
+                            &operation, steps[i],
+                            inputs[i]->x, inputs[i]->len ), statuses[i] );
+                        break;
+                    case 1: // input key
+                        psa_set_key_usage_flags( &attributes2, PSA_KEY_USAGE_DERIVE );
+                        psa_set_key_algorithm( &attributes2, alg );
+                        psa_set_key_type( &attributes2, PSA_KEY_TYPE_DERIVE );
+
+                        // other secret of type RAW_DATA passed with input_key
+                        if ( statuses[i] == PSA_ERROR_INVALID_ARGUMENT )
+                            psa_set_key_type( &attributes2, PSA_KEY_TYPE_RAW_DATA );
+
+                        PSA_ASSERT( psa_import_key( &attributes2,
+                            inputs[i]->x, inputs[i]->len,
+                            &keys[i] ) );
+
+                        TEST_EQUAL( psa_key_derivation_input_key(
+                                        &operation, steps[i], keys[i] ), statuses[i] );
+                        break;
+                    case 2: // key agreement
+                        psa_set_key_usage_flags( &attributes3, PSA_KEY_USAGE_DERIVE );
+                        psa_set_key_algorithm( &attributes3, alg );
+                        psa_set_key_type( &attributes3, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1) );
+
+                        PSA_ASSERT( psa_import_key( &attributes3,
+                            inputs[i]->x, inputs[i]->len,
+                            &keys[i] ) );
+
+                        TEST_EQUAL( psa_key_derivation_key_agreement(
+                                        &operation,
+                                        PSA_KEY_DERIVATION_INPUT_OTHER_SECRET,
+                                        keys[i], key_agreement_peer_key->x,
+                                        key_agreement_peer_key->len ), statuses[i] );
+                        break;
+                    case 3: // raw key agreement
```

Comment:
Ok, sorry if this is a stupid question, but I don't really get the testing value of that extra case: it uses `input_bytes()` as well, and I don't think it matters whether those bytes come from the output of `psa_raw_key_agreement()` or test data, does it?

(I get that I suggested this is how we would be doing this in TLS for now, but I don't think that implies it needs to be part of unit testing for the core function. We'll have TLS integration testing anyway.)

On one hand, I'm tempted to say extra testing never hurts, but OTOH it adds a bit of complexity (extra code, larger test matrix)...

Reply:
Yes, this is the same case as `input_bytes()`, but this is the use case with `psa_raw_key_agreement()`.
Can be removed.